### ISSUE TITLE
Moving from MySQL to SQLite

### ISSUE THREAD

**Issue by saschafoerster** [timeoff-management/application#317]
I tried to update timeout-management from an older version to the newest one (I can't find the older version). 

The older version was running in productive mode on MySQL:  /config/db.json
```

{
  "development": {
    "dialect": "sqlite",
    "storage": "./db.development.sqlite"
  },
  "test": {
    "username": "root",
    "password": null,
    "database": "database_test",
    "host": "127.0.0.1",
    "dialect": "mysql"
  },
  "production": {
    "username": "timeoff",
    "password": "#####",
    "database": "timeoff",
    "host": "localhost",
    "dialect": "mysql"
  }
}

```
After the update nothing was working, it seemed to me like MySQL-support was ditched. Is there any easy way to migrate the data? Or is the only way, export and add all users and holidays manually after updating?

**Comment by vpp** [comment]
Hi Sascha,

Did you run db-update script as per https://github.com/timeoff-management/application#updating-existing-instance-with-new-code ?

Thanks
Pavlo

**Comment by saschafoerster** [comment]
I did, but it didn't help, because the npm-packages for mysql where removed, so even the updates where adapted to to mysql-database, it couldn't connect to them afterwards.

**Comment by amirulahmad** [comment]
I tried to run the migration and got this error:

```
code: 'ER_CANNOT_ADD_FOREIGN',
     errno: 1215,
     sqlMessage: 'Cannot add foreign key constraint',
     sqlState: 'HY000',
     index: 0,
     sql: 'CREATE TABLE IF NOT EXISTS `user_allowance_adjustment` (`id` INTEGER NOT NULL auto_increment , `year` INTEGER NOT NULL DEFAULT \'2018\', `adjustment` INTEGER NOT NULL DEFAULT 0, `carried_over_allowance` INTEGER NOT NULL DEFAULT 0, `created_at` DATETIME NOT NULL, `user_id` INTEGER, PRIMARY KEY (`id`), FOREIGN KEY (`user_id`) REFERENCES `Users` (`id`) ON DELETE SET NULL ON UPDATE CASCADE) ENGINE=InnoDB;' }
```

..and the migration seems to run up to:
20171218-company-wide-message.js

**Comment by saschafoerster** [comment]
Problem solved. I found out, that I never used MySQL (even though I thought so). Everything was already saved in a SQLite file. So I could easily follow the update-path described in #179:

do fresh checkout from master on new place
copy DB file over to new place
copy config file over to new place (if you updated default values)
on new place "npm install"
on new place "npm run-script db-update"
and then finally "npm start"

This worked well (okay, I had to reset the password) and now I enjoy even more the great work from @vpp! Thanks!
